inst/tests/nafill.Rraw

```diff
@@ -186,3 +186,17 @@ if (test_bit64) {
   test(6.53, identical(coerceFill(-2147483648), list(NA_integer_, -2147483648,  as.integer64("-2147483648"))))
   test(6.54, identical(coerceFill(-2147483649), list(NA_integer_, -2147483649,  as.integer64("-2147483649"))))
 }
+
+# nan argument to treat NaN as NA in nafill, #4020
+x = c(NA, NaN, 0, 1, 2)
+ans1 = c(0, 0, 0:2)
+ans2 = c(0, NaN, 0:2)
+DT = data.table(a=x, b=x)
+test(7.1, nafill(x, fill=0), ans1)
+test(7.2, nafill(x, fill=0, nan=NaN), ans2)
+test(7.3, nafill(x, 'locf'), c(NA, NA, 0, 1:2))
```

Comment:
@jangorecki note this output -- `NaN` is converted to `NA` -- `nan=NA` means convert `NaN` to `NA` even if not replacing it. Makes sense?